plaso/engine/artifact_filters.py

```diff
@@ -75,71 +74,111 @@ def BuildFindSpecs(self, environment_variables=None):
+      environment_variables (list[EnvironmentVariableArtifact]):
+          environment variables.
+
+    Returns:
+      list[dfwinreg.FindSpec|dfvfs.FindSpec]: find specifications.
+    """
+    find_specs = []
+    for source in definition.sources:
+      if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+        for path_entry in set(source.paths):
+          specifications = self._BuildFindSpecsFromFileSource(
+              path_entry, source.separator, environment_variables,
+              self._knowledge_base.user_accounts)
+          find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+        for key_path in set(source.keys):
+          if self.CheckKeyCompatibility(key_path):
+            specifications = self._BuildFindSpecsFromRegistrySource(key_path)
+            find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+        # TODO: Handle Registry Values Once Supported in dfwinreg.
+        # https://github.com/log2timeline/dfwinreg/issues/98
+        logger.warning((
+            'Windows Registry values are not supported, extracting key: '
+            '"{0!s}"').format(source.key_value_pairs))
+
+        source_key_paths = {
+            key_value['key'] for key_value in source.key_value_pairs}
+        for key_path in source_key_paths:
+          if self.CheckKeyCompatibility(key_path):
+            specifications = self._BuildFindSpecsFromRegistrySource(key_path)
+            find_specs.extend(specifications)
+
+      elif (source.type_indicator ==
+            artifact_types.TYPE_INDICATOR_ARTIFACT_GROUP):
+        for name in source.names:
+          specifications = self._BuildFindSpecsFromGroupSource(
+              name, environment_variables)
+          find_specs.extend(specifications)
+
+      else:
+        logger.warning(
+            'Unsupported artifact definition source type: "{0:s}"'.format(
+                source.type_indicator))
+
+      return find_specs
+
+  def _BuildFindSpecsFromGroupSource(self, group_name, environment_variables):
+    """Builds find specifications from a artifact group source type.
+
+    Args:
+      group_name (str): artifact group name.
+      environment_variables (list[str]): environment variable attributes used to
+          dynamically populate environment variables in file and registry
+          artifacts.
+
+    Returns:
+      list[dfwinreg.FindSpec|dfvfs.FindSpec]: find specifications or None if no
+          artifact with the given name can be retrieved.
+    """
+    definition = self._artifacts_registry.GetDefinitionByName(group_name)
+    if not definition:
+      return None
+    return self._BuildFindSpecsFromArtifact(definition, environment_variables)
+
 
-  def BuildFindSpecsFromFileArtifact(
+  def _BuildFindSpecsFromFileSource(
```

Comment:
Technically this method neither uses a file artifact or artifact file source. Maybe:

`_BuildFindSpecsFromPath` or `_BuildFindSpecsFromFileSystemPath`

Reply:
Renamed to _BuildFindSpecsFromFileSourcePath to match the artifact source type naming: https://github.com/ForensicArtifacts/artifacts/blob/master/artifacts/source_type.py#L178